applications/GeoMechanicsApplication/custom_processes/reset_displacement_process.cpp

```diff
@@ -0,0 +1,75 @@
+// KRATOS___
+//     //   ) )
+//    //         ___      ___
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//  Main authors:    Richard Faasse
+//
+
+#include "reset_displacement_process.h"
+#include "includes/initial_state.h"
+#include "includes/kratos_parameters.h"
+#include "includes/model_part.h"
+#include "includes/ublas_interface.h"
+#include "includes/variables.h"
+
+#include <vector>
+
+namespace Kratos
+{
+
+ResetDisplacementProcess::ResetDisplacementProcess(ModelPart& rModelPart, const Parameters&)
+    : mrModelPart(rModelPart)
+{
+}
+
+void ResetDisplacementProcess::ExecuteInitialize()
+{
+    block_for_each(mrModelPart.Elements(), [this](Element& rElement) {
+        std::vector<Vector> stresses_on_integration_points;
+        rElement.CalculateOnIntegrationPoints(PK2_STRESS_VECTOR, stresses_on_integration_points,
```

Comment:
As discussed, for now the PK2_STRESS_VECTOR is the right choice for structural elements (please correct me if I'm wrong @WPK4FEM)